### PR TITLE
[release/6.0] Improve SSL platform detection (#64923)

### PR DESCRIPTION
**Pull request by wfurt**
manual port of #64923 to improve TLS detection on Azure images of Server 2022. 
(test only change) 

* CI matrix change: add Windows Server 2022

* Add registry check for ssl3 - tls1.2 tests

* Disable TLS1.3 on framework code

* Fix typo

* revert queue change

Co-authored-by: Jan Jahoda <jajahoda@microsoft.com>

### CODE REVIEW

**Comment by ericstj** [comment]
@wfurt is this ready to be merged?  Can we get fresh CI results and a signoff?

**Comment by carlossanlop** [comment]
@wfurt @dotnet/ncl this is an infra change targeting release/6.0. Today is the deadline for merging servicing changes to 6.0. Can someone take a look and confirm it's good to merge and sign off? Otherwise we'll have to wait for the next servicing release.

**Comment by wfurt** [comment]
This impacts tests. So it depends how much we care about failures. I feel this is ready but my team is probably all off already.

**Review by carlossanlop — APPROVED** [review]

**Comment by carlossanlop** [comment]
Ah my bad, I didn't notice it was test-only. I can sign it off. The CI passed successfully. I'll merge it now.

**Pull request merged**
